### ISSUE TITLE
lodepng for LIBX11-DEV library ?

### ISSUE THREAD

**Issue by spartrekus** [lvandeve/lodepng#71]
Groetjes,

The cool thing would be to avoid SDL and OPENGL for LODEPNG, as well C++ (g++, cmake,...).

Would it be possible to give a LODEPNG for X11 and plain C (if possible, lot of work I know,...)?

thank you for your efforts creating light libraries !!!

**Comment by stinos** [comment]
Not sure what you mean? This library doesn't depend on SDL nor openGL, and can be compiled with a plain C compiler.

**Comment by lvandeve** [comment]
Yep, it does support plain C90. There is an ifdef for C++ and for C in the same file, you can rename the .cpp to .c. It is just integrated all in one source file to keep it a single file library :)

**Comment by spartrekus** [comment]
The good thing would be that there is X11 (without SDL or Opengl). Just show a picture using the basic Xorg X11 library.

**Comment by spartrekus** [comment]
The problem is that since sdl 2.0 and higher versions, the graphical applications may run slow on many hardwares.

Really x11 is surely not very fancy, beautiful, but it is just working (reliable).

**Comment by lvandeve** [comment]
Well but the SDL example is just an example, if you want fast rendering it's better to use a good rendering engine anyway. Like, if X11 suits your needs, why not that one indeed.

In a real world application where speed matters, I assume you want to do more than just display a single PNG image in a window that closes as soon as you press any key, so there will be more work needed anyway than what a simple example in a PNG decoding library can give.
